connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -52,13 +52,18 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
     session.withActive {
 
       // Add debug information to the query execution so that the jobs are traceable.
-      val debugString = v.toString
-      session.sparkContext.setLocalProperty(
-        "callSite.short",
-        s"Spark Connect - ${StringUtils.abbreviate(debugString, 128)}")
-      session.sparkContext.setLocalProperty(
-        "callSite.long",
-        StringUtils.abbreviate(debugString, 2048))
+      try {
```

Comment:
```
Started distributed training with 2 executor processes
java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOfRange(Arrays.java:3664)
	at java.lang.String.<init>(String.java:207)
	at java.lang.StringBuilder.toString(StringBuilder.java:407)
	at org.sparkproject.connect.protobuf.TextFormatEscaper.escapeBytes(TextFormatEscaper.java:112)
	at org.sparkproject.connect.protobuf.TextFormatEscaper.escapeBytes(TextFormatEscaper.java:119)
	at org.sparkproject.connect.protobuf.TextFormat.escapeBytes(TextFormat.java:2364)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:593)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printSingleField(TextFormat.java:752)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printField(TextFormat.java:457)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printMessage(TextFormat.java:714)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.print(TextFormat.java:367)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:606)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printSingleField(TextFormat.java:752)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printField(TextFormat.java:457)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printMessage(TextFormat.java:714)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.print(TextFormat.java:367)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:606)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printSingleField(TextFormat.java:752)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printField(TextFormat.java:457)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printMessage(TextFormat.java:714)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.print(TextFormat.java:367)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:606)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printSingleField(TextFormat.java:752)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printField(TextFormat.java:457)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printMessage(TextFormat.java:714)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.print(TextFormat.java:367)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:606)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printSingleField(TextFormat.java:752)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printField(TextFormat.java:457)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printMessage(TextFormat.java:714)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.print(TextFormat.java:367)
	at org.sparkproject.connect.protobuf.TextFormat$Printer.printFieldValue(TextFormat.java:606)
Extracting /home/ruifeng.zheng/spark/python/target/50b3d81f-67a9-46de-9beb-59b733c16e54/tmp72g322ys/MNIST/raw/t10k-labels-idx1-ubyte.gz to /home/ruifeng.zheng/spark/python/target/50b3d81f-67a9-46de-9beb-59b733c16e54/tmp72g322ys/MNIST/raw
```

`v.toString` is keeping OOM in `TorchDistributorDistributedUnitTestsOnConnect`.
This OOM is related to OS or Java Version, it was thrown in Linux+Java8, but doesn't emerge in my local env (macos+java11).


The GA resources for free usage is limited to 2U 6G (confirmed with @Yikun), and I can not allocate enough driver memory for distributed pytorch training without this fix.

Reply:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Yes, 2U7G, and taking into account the system occupancy, 2U 6+G available:
https://github.com/apache/spark/blob/f541301b7680d96611796d92943d4ec72c71ec0d/.github/workflows/build_and_test.yml#L1028-L1029